extensions/amp-auto-lightbox/0.1/amp-auto-lightbox.js

```diff
@@ -114,13 +127,70 @@ export class Criteria {
 
   /**
    * @param {!Element} element
-   * @return {boolean}
+   * @return {!Promise<boolean>}
    */
   static meetsAll(element) {
-    return Criteria.meetsSizingCriteria(element) &&
-      Criteria.meetsTreeShapeCriteria(element);
+    if (!Criteria.meetsSimpleCriteria(element) ||
+        !Criteria.meetsTreeShapeCriteria(element)) {
+      return resolveFalse();
+    }
+    return Criteria.meetsComplexCriteria(element);
   }
 
+  /**
+   * Criteria that is "simple", ie runs quickly and discards elements in order
+   * to shortcircuit.
+   * @param {!Element} element
+   * @return {boolean}
+   */
+  static meetsSimpleCriteria(element) {
+    switch (element.tagName) {
```

Comment:
Is there a benefit of the `switch` vs an `if`? I feel like `switch` would only make more sense if there were additional `tagName` cases to switch on later?